crash/panic - okta_user_schema_property

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version
```
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/okta/okta v3.22.1
```

### Affected Resource(s)


<!--- Please list the affected resources and data sources. --->

- okta_user_schema_property

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

```
╷
│ Error: Plugin did not respond
│
│   with module.auth_server.okta_auth_server_claim.resource_claims["my_var_name"],
│   on modules/auth_server/main.tf line 201, in resource "okta_auth_server_claim" "resource_claims":
│  201: resource "okta_auth_server_claim" "resource_claims" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.auth_server.okta_auth_server_claim.resource_claims["threat_access_roles"],
│   on modules/auth_server/main.tf line 201, in resource "okta_auth_server_claim" "resource_claims":
│  201: resource "okta_auth_server_claim" "resource_claims" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.user_schema.okta_user_schema_property.my_var_name,
│   on modules/user_schema/schema.tf line 1, in resource "okta_user_schema_property" "my_var_name":
│    1: resource "okta_user_schema_property" "my_var_name" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-okta_v3.22.1 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 69 [running]:
github.com/okta/terraform-provider-okta/okta.buildEnum({0xc000282900, 0x4, 0x0}, {0xc00041a460, 0x6})
	github.com/okta/terraform-provider-okta/okta/user_schema_property.go:313 +0x2a9
github.com/okta/terraform-provider-okta/okta.buildUserCustomSchemaAttribute(0xc00041a250)
	github.com/okta/terraform-provider-okta/okta/user_schema_property.go:367 +0x1cd
github.com/okta/terraform-provider-okta/okta.resourceUserSchemaCreateOrUpdate({0x1b0e108, 0xc00051c300}, 0xc0007936c0, {0x18afa80, 0xc000242270})
	github.com/okta/terraform-provider-okta/okta/resource_okta_user_custom_schema_property.go:112 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0004500e0, {0x1b0e098, 0xc000664000}, 0x2, {0x18afa80, 0xc000242270})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:341 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0004500e0, {0x1b0e098, 0xc000664000}, 0xc000468b60, 0xc000122600, {0x18afa80, 0xc000242270})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000440f00, {0x1b0e098, 0xc000664000}, 0xc00007eb90)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000284e80, {0x1b0e140, 0xc000108270}, 0xc0002a4380)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1988f40, 0xc000284e80}, {0x1b0e140, 0xc000108270}, 0xc000048480, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00029aa80, {0x1b1dd70, 0xc000198180}, 0xc00054e000, 0xc0004b1680, 0x20b6980, 0x0)
	google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00029aa80, {0x1b1dd70, 0xc000198180}, 0xc00054e000, 0x0)
	google.golang.org/grpc@v1.41.0/server.go:1608 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.41.0/server.go:923 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.41.0/server.go:921 +0x294

Error: The terraform-provider-okta_v3.22.1 plugin crashed!
```

### Expected Behavior

Would expect to be able to destroy and add custom profile attributes at will.

### Actual Behavior

The provider destroyed this custom attribute, when it was re-added the panic is consistently encountered. 
The okta instance also now appears to be in an unstable state. Adding the same variable name as a custom attribute using the admin console will cause an unspecified "Internal Server Error"

![image](https://user-images.githubusercontent.com/470759/162053601-78cc5b72-53f0-4c19-bca9-993ab55eede3.png)

### Steps to Reproduce

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

- #0000

--- a/okta/user_schema_property.go
+++ b/okta/user_schema_property.go
@@ -293,29 +293,6 @@ func buildOneOf(ae []interface{}, elemType string) ([]*okta.UserSchemaAttributeE
 	return oneOf, nil
 }
 
-func buildEnum(ae []interface{}, elemType string) ([]interface{}, error) {
-	enum := make([]interface{}, len(ae))
-	for i := range ae {
-		switch elemType {
-		case "number":
-			f, err := strconv.ParseFloat(ae[i].(string), 64)
-			if err != nil {
-				return nil, errInvalidElemFormat
-			}
-			enum[i] = f
-		case "integer":
-			f, err := strconv.Atoi(ae[i].(string))
-			if err != nil {
-				return nil, errInvalidElemFormat
-			}
-			enum[i] = f
-		default:
-			enum[i] = ae[i].(string)
-		}
-	}
-	return enum, nil
-}
-
 func flattenEnum(enum []interface{}) []interface{} {
 	result := make([]interface{}, len(enum))
 	for i := range enum {

--- a/okta/utils.go
+++ b/okta/utils.go
@@ -6,6 +6,7 @@ import (
 	"errors"
 	"fmt"
 	"net/http"
+	"strconv"
 	"strings"
 	"unicode"
 
@@ -340,4 +341,40 @@ func validatePriority(in, out int64) error {
 		return fmt.Errorf("provided priority was not valid, got: %d, API responded with: %d. See schema for attribute details", in, out)
 	}
 	return nil
+}
+
+func buildEnum(ae []interface{}, elemType string) ([]interface{}, error) {
+	enum := make([]interface{}, len(ae))
+	for i, aeVal := range ae {
+		if aeVal == nil {
+			switch elemType {
+			case "number":
+				enum[i] = float64(0)
+			case "integer":
+				enum[i] = 0
+			default:
+				enum[i] = ""
+			}
+			continue
+		}
+
+		aeStr := aeVal.(string)
+		switch elemType {
+		case "number":
+			f, err := strconv.ParseFloat(aeStr, 64)
+			if err != nil {
+				return nil, errInvalidElemFormat
+			}
+			enum[i] = f
+		case "integer":
+			f, err := strconv.Atoi(aeStr)
+			if err != nil {
+				return nil, errInvalidElemFormat
+			}
+			enum[i] = f
+		default:
+			enum[i] = aeStr
+		}
+	}
+	return enum, nil
 }

--- a/okta/utils.go
+++ b/okta/utils.go
@@ -358,7 +358,10 @@ func buildEnum(ae []interface{}, elemType string) ([]interface{}, error) {
 			continue
 		}
 
-		aeStr := aeVal.(string)
+		aeStr, ok := aeVal.(string)
+		if !ok {
+			return nil, fmt.Errorf("expected %+v value to cast to string", aeVal)
+		}
 		switch elemType {
 		case "number":
 			f, err := strconv.ParseFloat(aeStr, 64)

--- a/okta/utils_test.go
+++ b/okta/utils_test.go
@@ -1,6 +1,7 @@
 package okta
 
 import (
+	"reflect"
 	"strings"
 	"testing"
 
@@ -87,6 +88,63 @@ func TestBuildSchema(t *testing.T) {
 		if actual != test.expected {
 			t.Errorf("buildSchema test failed, test maps: %v, elements %s, expected %v, actual %v",
 				testMaps, test.element, test.expected, actual)
+		}
+	}
+}
+
+func TestBuildEnum(t *testing.T) {
+	tests := []struct {
+		name        string
+		ae          []interface{}
+		elemType    string
+		expected    []interface{}
+		shouldError bool
+	}{
+		{
+			"number slice including empty value",
+			[]interface{}{"1.1", nil},
+			"number",
+			[]interface{}{1.1, 0.0},
+			false,
+		},
+		{
+			"integer slice including empty value",
+			[]interface{}{"1", nil},
+			"integer",
+			[]interface{}{1, 0},
+			false,
+		},
+		{
+			"string slice including empty value",
+			[]interface{}{"one", nil},
+			"string",
+			[]interface{}{"one", ""},
+			false,
+		},
+		{
+			"number slice with invalid value and empty value",
+			[]interface{}{"one", nil},
+			"number",
+			nil,
+			true,
+		},
+		{
+			"integer slice with invalid value and empty value",
+			[]interface{}{"one", nil},
+			"integer",
+			nil,
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		actual, err := buildEnum(test.ae, test.elemType)
+		if test.shouldError && err == nil {
+			t.Errorf("%q - buildEnum should have errored on %+v, %s, got %+v", test.name, test.ae, test.elemType, actual)
+
+		}
+		if !reflect.DeepEqual(test.expected, actual) {
+			t.Errorf("%q - buildEnum expected %+v, got %+v", test.name, test.expected, actual)
 		}
 	}
 }

--- a/okta/utils_test.go
+++ b/okta/utils_test.go
@@ -135,6 +135,13 @@ func TestBuildEnum(t *testing.T) {
 			nil,
 			true,
 		},
+		{
+			"ae slice is not string slice",
+			[]interface{}{1, 2, 3},
+			"integer",
+			nil,
+			true,
+		},
 	}
 
 	for _, test := range tests {

